From 3977d176269ff5d9cf7be6270f470f06794f9dda Mon Sep 17 00:00:00 2001
From: Stefan Hansson <newbyte@postmarketos.org>
Date: Sat, 30 Sep 2023 17:31:45 +0200
Subject: [PATCH] thumbnails: Fix compilation on 32-bit musl

%lu probably works in most cases, but on 32-bit musl (not sure
about glibc) it complains about that %lu expects
"long unsigned int" but time_t is "long long int". As such, work
around this by copying what gnome-desktop does for this case.
---
 src/nautilus-thumbnails.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/nautilus-thumbnails.c b/src/nautilus-thumbnails.c
index 9d6ade04f..04d637066 100644
--- a/src/nautilus-thumbnails.c
+++ b/src/nautilus-thumbnails.c
@@ -481,7 +481,8 @@ thumbnail_generated_cb (GObject      *source_object,
 
     if (pixbuf != NULL)
     {
-        g_autofree gchar *mtime = g_strdup_printf ("%lu", info->updated_file_mtime);
+        g_autofree gchar *mtime = g_strdup_printf ("%" G_GINT64_FORMAT,
+                                                   (gint64) info->updated_file_mtime);
 
         DEBUG ("(Thumbnail Async Thread) Saving thumbnail: %s\n",
                info->image_uri);
-- 
2.42.0

